source/android/mobile/src/main/java/io/adaptivecards/adaptivecardssample/CustomObjects/CardElements/CustomInput.java

```diff
@@ -0,0 +1,127 @@
+            super(baseInputElement);
+            m_editText = editText;
+        }
+
+        @Override
+        public String getInput()
+        {
+            return m_editText.getText().toString();
+        }
+
+        @Override
+        public void setInput(String input)
+        {
+            m_editText.setText(input);
+        }
+
+        private EditText m_editText = null;
+    }
+
+    public static class CustomInputRenderer extends BaseCardElementRenderer
+    {
+        @Override
+        public View render(RenderedAdaptiveCard renderedCard,
+                           Context context,
+                           FragmentManager fragmentManager,
+                           ViewGroup viewGroup,
+                           BaseCardElement baseCardElement,
+                           ICardActionHandler cardActionHandler,
+                           HostConfig hostConfig,
+                           RenderArgs renderArgs) throws Exception
+        {
+            EditText renderedInput = new EditText(context);
+            renderedInput.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
+
+            renderedInput.setBackgroundColor(Color.GREEN);
+            renderedInput.setTag(new TagContent(baseCardElement));
+            viewGroup.addView(renderedInput);
+
+            BaseInputElement inputElement = Util.castTo(baseCardElement, BaseInputElement.class);
+            renderedCard.registerInputHandler(new CustomInputHandler(inputElement, renderedInput), renderArgs.getContainerCardId());
+            return renderedInput;
+        }
+    }
+
+
+    public static class CustomInputParser extends BaseCardElementParser
+    {
+        @Override
+        public BaseCardElement Deserialize(ParseContext context, JsonValue value)
+        {
+            CustomInput element = new CustomInput(CardElementType.Custom);
+            element.SetElementTypeString(customInputTypeString);
+            String val = value.getString();
+            try
+            {
+                JSONObject obj = new JSONObject(val);
+                element.SetId(obj.getString("id"));
+                element.SetLabel(obj.getString("label"));
+            }
+            catch (JSONException e)
+            {
+                e.printStackTrace();
+            }
+            return element;
+        }
+
+        @Override
+        public BaseCardElement DeserializeFromString(ParseContext context, String jsonString)
+        {
+            CustomInput element = new CustomInput(CardElementType.Custom);
+            element.SetElementTypeString(customInputTypeString);
+            try
+            {
+                JSONObject obj = new JSONObject(jsonString);
+                element.SetId(obj.getString("id"));
+                element.SetLabel(obj.getString("label"));
+            }
+            catch (JSONException e)
+            {
+                e.printStackTrace();
+            }
```

Comment:
Is that right? If we can't parse the input json for the custom input element, do we just return defaults? If so this is yet another place where we should both annotate the sample and in case this is what we are doing for rendering native elements, update our documentation.

Reply:
the native elements parse in C++ and as so they throw a different type of exception, for samples we try to keep the behaviour simple, that's why we print the exception trace